src/scripts/main.js

```diff
@@ -1,69 +1,197 @@
+    country: 'Malaysia',
+  },
+  {
+    name: 'Black Penny Villas',
+    city: 'BTDC, Nuca Dua',
     country: 'Indonesia',
-    city: 'Bali',
-    hotel: 'Ubud Bali Resort&SPA',
   },
   {
+    name: 'Koko Hotel',
+    city: 'Tokyo',
+    country: 'Japan',
+  },
+  {
+    name: 'Ramada Plaza',
+    city: 'Istanbul',
+    country: 'Turkey',
+  },
+  {
+    name: 'Waikiki Resort Hotel',
+    city: 'Hawaii',
+    country: 'USA',
+  },
+  {
+    name: 'Puro Hotel',
+    city: 'Krakow',
+    country: 'Poland',
+  },
+  {
+    name: 'Asma Suites',
+    city: 'Santorini',
+    country: 'Greece',
+  },
+  {
+    name: 'Cityden Apartments',
+    city: 'Amsterdam',
     country: 'Netherlands',
-    city: 'Rotterdam',
-    hotel: 'King Kong Hostel',
   },
   {
-    country: 'Marocco',
-    city: 'Ourika',
-    hotel: 'Rokoko Hotel',
+    name: 'Mandarin Oriental',
+    city: 'Miami',
+    country: 'USA',
+  },
+  {
+    name: 'Concept Terrace Hotel',
+    city: 'Rome',
+    country: 'Italy',
+  },
+  {
+    name: 'Ponta Mar Hotel',
+    city: 'Fortaleza',
+    country: 'Brazil',
+  },
+  {
+    name: 'Four Seasons Hotel',
+    city: 'Sydney',
+    country: 'Australia',
+  },
+  {
+    name: 'Le Meridien',
+    city: 'Nice',
+    country: 'France',
   },
   {
+    name: 'Apart Neptun',
+    city: 'Gdansk',
+    country: 'Poland',
+  },
+  {
+    name: 'Lux Isla',
+    city: 'Ibiza',
+    country: 'Spain',
+  },
+  {
+    name: 'Nox Hostel',
+    city: 'London',
+    country: 'UK',
+  },
+  {
+    name: 'Leonardo Vienna',
+    city: 'Vienna',
+    country: 'Austria',
+  },
+  {
+    name: 'Adagio Aparthotel',
+    city: 'Edinburgh',
+    country: 'UK',
+  },
+  {
+    name: 'Steigenberger Hotel',
+    city: 'Hamburg',
     country: 'Germany',
-    city: 'Berlin',
-    hotel: 'Hotel Rehberge Berlin Mitte',
   },
 ];
 
+const palindrome = function (word) {
+  return word === word.split('').reverse().join('');
+};
+
+const myword = 'шалаш';
+console.log(palindrome(myword));
+
 const find = 'germany';
 
 const re2 = new RegExp(`${find}`, 'i');
 
-const search = function (str) {
-  const result = [];
-  for (let i = 0; i < data.length; i++) {
-    const obj = `${data[i].country}, ${data[i].city}, ${data[i].hotel}`;
-    if (obj.match(str) !== null) {
-      result[result.length] = obj.match(str).input;
-    }
+const filterItems = hotels.reduce((result, item) => {
+  const str = `${item.country}, ${item.city}, ${item.name}`;
+  if (str.match(re2) !== null) {
+    result.push(str.match(re2).input);
   }
   return result;
+}, []);
+
+console.log(filterItems);
+
+// __________________________________________________________________________________
+
+const tmpArray = [];
+const tmpObject = [];
+
+const uniqueCountry = (arr) => {
+  arr.forEach((item) => {
+    if (tmpArray.indexOf(item.country) === -1) {
+      tmpArray.push(item.country, item.city);
+      tmpObject.push({ [item.country]: [item.city] });
+    } else {
+      tmpObject.item.country.push(item.city);
```

Comment:
Look at your array tmpObject `v`
```
[ { "Russia" : [ "Moscow" ] }, { ... }, { ... } ... ]
```
We can't get an internal array with "Moscow" by `tmpObject.item.country`. And even by `tmpObject[item.country]`. We need to find the object  `{ "Russia" : [ "Moscow" ] }` in tmpObject.